src/backend/distributed/commands/table.c

```diff
@@ -507,6 +508,12 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		return PreprocessAlterTableStmt(node, alterTableCommand, processUtilityContext);
 	}
 
+	if (DropsForeignKey(leftRelationId, alterTableStatement))
+	{
+		/* invalidating the fkey graph since cached foreign keys could cause crashes */
+		InvalidateForeignKeyGraph();
```

Comment:
since we are in preprocess, we should use `MarkInvalidateForeignKeyGraph` so that we don't invalidate the graph when ddl fails